### PR TITLE
Workbook descriptor schema

### PR DESCRIPTION
**Pull request by AnonymousWalker**
This change is [Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752)

### CODE REVIEW

File: common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/WorkbookDatabaseAccessor.kt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import org.wycliffeassociates.otter.common.data.primitives.ContentType
+import org.wycliffeassociates.otter.common.data.primitives.Language
+import org.wycliffeassociates.otter.common.data.primitives.ResourceMetadata
+import org.wycliffeassociates.otter.common.data.workbook.DateHolder
+import org.wycliffeassociates.otter.common.data.workbook.Translation
+import org.wycliffeassociates.otter.common.domain.collections.UpdateTranslation
+
+typealias ModelTake = org.wycliffeassociates.otter.common.data.primitives.Take
+
+interface IWorkbookDatabaseAccessors {
```
Comment (AnonymousWalker) [inline common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/WorkbookDatabaseAccessor.kt:19]: moved from WorkbookRepository

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752)* status: 0 of 24 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker)

___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IWorkbookDescriptorRepository.kt` line 8 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752#-NWHpv3zDaDT50n73jwQ:-NWHpv3zDaDT50n73jwR:b-y6tmna) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/f3ad7a311ad1e3ca6df67ab07b7ca32b24489a35/common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IWorkbookDescriptorRepository.kt#L8)):*
> ```Kotlin
> import org.wycliffeassociates.otter.common.data.workbook.WorkbookDescriptor
> 
> interface IWorkbookDescriptorRepository {
> ```

Because these describe Workbooks, (it's like Workbook Lite™) is there a reason this should have its own repository?
___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/DatabaseMigrator.kt` line 340 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752#-NWHqIr05t513pfrmnCr:-NWHqIr05t513pfrmnCs:bc6b1ja) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/f3ad7a311ad1e3ca6df67ab07b7ca32b24489a35/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/DatabaseMigrator.kt#L340)):*
> ```Kotlin
>     }
> 
>     private fun migrate11to12(dsl: DSLContext, current: Int): Int {
> ```

We've always been documenting what the migration does, could you include that here? See migration functions above for examples.

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752)* status: 0 of 24 files reviewed, 3 unresolved discussions (waiting on @jsarabia)

___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IWorkbookDescriptorRepository.kt` line 8 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752#-NWHpv3zDaDT50n73jwQ:-NWI7klu-6mdmQH-bw1K:b9tz03u) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/f3ad7a311ad1e3ca6df67ab07b7ca32b24489a35/common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IWorkbookDescriptorRepository.kt#L8)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Because these describe Workbooks, (it's like Workbook Lite™) is there a reason this should have its own repository?
</blockquote></details>

in order to access the database through the DAOs, something must be located under the jvm package since the WorkbookRepository and its DatabaseAccessor are in the common package. These classes don't have direct access to the database.

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752)* status: 0 of 24 files reviewed, 3 unresolved discussions (waiting on @jsarabia)

___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/DatabaseMigrator.kt` line 340 at r2](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752#-NWHqIr05t513pfrmnCr:-NWIKKg60YzMfr0xP_n6:b-896fix) ([raw file](https://github.com/Bible-Translation-Tools/Orature/blob/f3ad7a311ad1e3ca6df67ab07b7ca32b24489a35/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/DatabaseMigrator.kt#L340)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

We've always been documenting what the migration does, could you include that here? See migration functions above for examples.
</blockquote></details>

Done.

**Review by jsarabia — APPROVED** [review]
:lgtm:

> Reviewed 2 of 24 files at r1.
> *[Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/752#-:-NWiHox7CundGPKZmHEy:bnfp4nl)* status: 2 of 24 files reviewed, 1 unresolved discussion

**Pull request merged**